documentation/design/proposals/console-1.0-backend.adoc

```diff
@@ -0,0 +1,117 @@
+:toc:
+:toclevels: 3
+
+# Proposal - Console changes to adopt to 1.0
+
+:toc-placement!:
+toc::[]
+
+=== Scope of this document
+
+* Console changes related to 1.0 shared infrastructure - IoT is covered separately
+* Features available in 1.0
+* Features not available in 1.0
+* Changes required to console APIs for 1.0
+* Changes required to console backend for 1.0
+* Changes required to console UIs for 1.0
+
+The changes in document are subject to ... change! There are several types related to authentication and authorization that are not yet implemented. The design for these will come at a later stage.
+
+=== Features
+
+For 1.0, the set of EnMasse resources that the user must be able to maintain from the UI is:
+
+* MessagingTenant
+* MessagingAddress
+* MessagingEndpoint
+* MessagingUser
+* MessagingRole
+* MessagingRoleBinding
+* IoTProject
+
+Given the timelines, service admin console should not be in scope for 1.0
+
+=== Console API/UI changes
+
+==== AddressSpace => MessagingTenant
+
+image::console_1.0_overview.png[]
+
+* `AddressSpace` is removed and replaced by `MessagingTenant`, and there can now only be a single MessagingTenant named 'default' per Kubernetes namespace. Most of the properties like plan and authentication service (now identity provider) is now optional, and the minimum information needed to create a tenant is its namespace. In addition, a tenant may have a set of capabilities such as 'transactional'.
+
```

Comment:
When in the UI will the user find the IoTProjects?

Reply:
the IoTProjects will be displayed in the same list as the MessagingTenants